src/app/core/entity-components/entity-select/entity-select/entity-select.component.ts

```diff
@@ -0,0 +1,241 @@
+   * <br>
+   * <pre>
+   *   <ng-template
+   *      #myTemplate
+   *      let-entity="entity">
+   *      {{entity.name}}
+   *   </ng-template>
+   * </pre>
+   * this would then be passed via <code>[entityView]="myTemplate"</code>
+   * and would cause all entities that are to be displayed to simply display their names
+   */
+  @Input() entityView: TemplateRef<any>;
+  /**
+   * The view used to render autocomplete-options.
+   * This has the same behavior as {@link entityView}
+   */
+  @Input() autocompleteView: TemplateRef<any>;
+  /**
+   * true when this is loading and false when it's ready.
+   * This subject's state reflects the actual loading resp. the 'readiness'-
+   * state of this component. Will trigger once loading is done
+   */
+  loading = new BehaviorSubject(true);
+  @Input() disabled: boolean = false;
+
+  allEntities: E[] = [];
+
+  filteredEntities: Observable<E[]>;
+  @ViewChild("inputField") inputField: ElementRef<HTMLInputElement>;
+
+  formControl = new FormControl();
+
+  readonly separatorKeysCodes: number[] = [ENTER, COMMA];
+
+  constructor(
+    private entityMapperService: EntityMapperService,
+    private loggingService: LoggingService
+  ) {
+    this.filteredEntities = this.formControl.valueChanges.pipe(
+      filter((value) => value === null || typeof value === "string"), // sometimes produces entities
+      map((searchText?: string) =>
+        searchText ? this._filter(searchText) : this.allEntities
+      ),
+      map((entities: E[]) =>
+        entities.filter(
+          (e: E) => !this.isSelected(e) && this.additionalFilter(e)
+        )
+      )
+    );
+  }
+  /**
+   * The accessor used for filtering and when selecting a new
+   * entity.
+   * <br> Per default, this filters for the name. If the entity
+   * has no name, this filters for the entity's id.
+   */
+  @Input() accessor: accessorFn<E> = (e) => e["name"] || e.getId();
+  /**
+   * Additional filters
+   */
+  @Input() additionalFilter: (T) => boolean = (_) => true;
+  /**
+   * selects a given entity and emits values
+   * @param entity the entity to select
+   */
+  selectEntity(entity: E) {
+    this._selection.push(entity);
+    this._emitChange();
+    this.inputField.nativeElement.value = "";
+    this.formControl.setValue(null);
+  }
+  /**
+   * called when a key code from {@link separatorKeysCodes}
+   * is recorded and the user has entered a new entity-name (resp.
+   * whatever the accessor defines)
+   * @param event the event to call this with
+   */
+  add(event: MatChipInputEvent) {
+    const value = event.value;
+
+    if (value) {
+      const entity = this.allEntities.find(
+        (e) => this.accessor(e) === value.trim()
+      );
+      if (entity) {
+        this.selectEntity(entity);
+      }
+    }
+  }
+
+  private _filter(value: string): E[] {
+    const filterValue = value.toLowerCase();
+    return this.allEntities.filter((entity) =>
+      this.accessor(entity).toLowerCase().startsWith(filterValue)
```

Comment:
The additional filter will only be applied to the entities that 'make' it through the original one.
This two-way filtering is needed since at the first stage either `_filter` is called, or all entities are being set (if the text is `null`). Moving additional filters here would mean that the filter only applies when a certain search-text is being set (would not apply when the selection is empty)

Reply:
I changed this so there is only one filter-function. This should make things more clear